### PR TITLE
fix --try-software-version using non-greedy matching + lookahead assertion, add test for tweak_one

### PR DESCRIPTION
**Pull request by boegel**
fix for #2163; https://github.com/hpcugent/easybuild-framework/issues/2163#issuecomment-286532957 is probably useful to get a good idea of why this fix is needed

### CODE REVIEW

File: easybuild/framework/easyconfig/tweak.py [hunk 1]
```diff
@@ -179,7 +179,9 @@ def __repr__(self):
     for (key, val) in tweaks.items():
 
         if isinstance(val, list):
-            regexp = re.compile(r"^(?P<key>\s*%s)\s*=\s*(?P<val>\[(.|\n)*\])\s*$" % key, re.M)
+            # use non-greedy matching for list value using '*?' to avoid including other parameters in match,
+            # and a lookahead assertion (?=...) so next line is either another parameter definition or a blank line
+            regexp = re.compile(r"^(?P<key>\s*%s)\s*=\s*(?P<val>\[(.|\n)*?\])\s*$(?=\n^(\w+\s*=.*|\s*)$)" % key, re.M)
```
Comment (wpoely86) [inline easybuild/framework/easyconfig/tweak.py]: why the last `|`?

Comment (wpoely86) [inline easybuild/framework/easyconfig/tweak.py]: Doesn't `\s` also match `\n` in multiline mode?

Comment (boegel) [inline easybuild/framework/easyconfig/tweak.py]: The last `|` in the lookahead assertion is there to check that the next line is either a parameter definition or a blank line (or the end of the file, see last commit).

Where is it relevant that `\s` also matches `\n`?

**Comment by boegel** [comment]
Thanks for the review @wpoely86!

**Pull request merged**